### PR TITLE
move CommCareCaseGroup out of case app

### PR DESCRIPTION
**Pull request by dannyroberts**
into new casegroups app

(wait for build)

`casexml.apps.case.models` didn't seem like the right place for `CommCareCaseGroup` (for example, if cases lived in some other database, I wouldn't expect `CommCareCaseGroup` to follow them there). In a later PR I'll move `case/groups_by_domain` to `casegroups/groups_by_domain` to decouple reindexing it from the case views as well.

### CODE REVIEW

File: corehq/apps/casegroups/models.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+    """
+        This is a group of CommCareCases. Useful for managing cases in larger projects.
+    """
+    name = StringProperty()
+    domain = StringProperty()
+    cases = ListProperty()
+    timezone = StringProperty()
+
+    def get_time_zone(self):
+        # Necessary for the CommCareCaseGroup to interact with CommConnect, as if using the CommCareMobileContactMixin
```
Comment (dimagimon) [inline corehq/apps/casegroups/models.py]: E501 line too long (118 > 115 characters)

File: corehq/apps/reminders/management/commands/migrate_surveysample_group_2013_09.py [hunk 2]
```diff
@@ -1,6 +1,7 @@
 from django.core.management.base import LabelCommand
 import sys
-from casexml.apps.case.models import CommCareCase, CommCareCaseGroup
+from casexml.apps.case.models import CommCareCase
```
Comment (dimagimon) [inline corehq/apps/reminders/management/commands/migrate_surveysample_group_2013_09.py]: F401 'CommCareCase' imported but unused

File: corehq/apps/reports/filters/select.py [hunk 3]
```diff
@@ -5,7 +5,8 @@
 from django.utils.translation import ugettext_noop, ugettext_lazy
 from django.utils.translation import ugettext as _
 
-from casexml.apps.case.models import CommCareCase, CommCareCaseGroup
+from casexml.apps.case.models import CommCareCase
```
Comment (dimagimon) [inline corehq/apps/reports/filters/select.py]: F401 'CommCareCase' imported but unused

File: corehq/ex-submodules/casexml/apps/case/models.py [hunk 4]
```diff
@@ -1026,3 +953,7 @@ def _type_sort(action_type):
     Consistent ordering for action types
     """
     return const.CASE_ACTIONS.index(action_type)
+
+
+# import signals
+import casexml.apps.case.signals
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/case/models.py]: F401 'casexml' imported but unused

File: corehq/apps/reports/filters/select.py [hunk 5]
```diff
@@ -4,10 +4,10 @@
 from django.conf import settings
 from django.utils.translation import ugettext_noop, ugettext_lazy
 from django.utils.translation import ugettext as _
+from corehq.apps.casegroups.dbaccessors import get_case_groups_in_domain
 
-from casexml.apps.case.models import CommCareCase, CommCareCaseGroup
+from corehq.apps.casegroups.models import CommCareCaseGroup
```
Comment (dimagimon) [inline corehq/apps/reports/filters/select.py]: F401 'CommCareCaseGroup' imported but unused

File: corehq/apps/reports/filters/select.py [hunk 6]
```diff
@@ -4,10 +4,10 @@
 from django.conf import settings
 from django.utils.translation import ugettext_noop, ugettext_lazy
 from django.utils.translation import ugettext as _
+from corehq.apps.casegroups.dbaccessors import get_case_groups_in_domain, \
```
Comment (dimagimon) [inline corehq/apps/reports/filters/select.py]: F401 'get_case_groups_in_domain' imported but unused

**Comment by dannyroberts** [comment]
@benrudolph, @kaapstorm buddy

**Pull request merged**